mne/io/fieldtrip/tests/test_fieldtrip.py

```diff
@@ -0,0 +1,206 @@
+                      pandas_not_found_warning_msg, get_raw_info, check_data,
+                      assert_warning_in_record)
+from ..utils import NOINFO_WARNING
+from mne.utils import _check_pandas_installed
+
+# missing: KIT: biggest problem here is that the channels do not have the same
+# names.
+# EGI: no calibration done in FT. so data is VERY different
+
+all_systems_raw = ['neuromag306', 'CTF', 'CNT', 'BTI', 'eximia']
+all_systems_epochs = ['neuromag306', 'CTF', 'CNT']
+all_versions = ['v7', 'v73']
+use_info = [True, False]
+all_test_params_raw = list(itertools.product(all_systems_raw, all_versions,
+                                             use_info))
+all_test_params_epochs = list(itertools.product(all_systems_epochs,
+                                                all_versions,
+                                                use_info))
+
+no_info_warning = {'expected_warning': RuntimeWarning,
+                   'message': NOINFO_WARNING}
+
+
+@testing.requires_testing_data
+@pytest.mark.filterwarnings('ignore::RuntimeWarning')
+@pytest.mark.parametrize('cur_system, version, use_info',
+                         all_test_params_epochs)
+def test_averaged(cur_system, version, use_info):
+    """Test comparing reading an Evoked object and the FieldTrip version."""
+    test_data_folder_ft = get_data_paths(cur_system)
+    mne_avg = get_averaged_data(cur_system)
+    if use_info:
+        info = get_raw_info(cur_system)
+        pytestwarning = {'expected_warning': None}
+    else:
+        info = None
+        pytestwarning = no_info_warning
+
+    cur_fname = os.path.join(test_data_folder_ft,
+                             'averaged_%s.mat' % (version,))
+    if version == 'v73' and not _has_h5py():
+        with pytest.raises(ImportError):
+            mne.io.read_evoked_fieldtrip(cur_fname, info)
+        return
+
+    with pytest.warns(**pytestwarning):
+        avg_ft = mne.io.read_evoked_fieldtrip(cur_fname, info)
+
+    mne_data = mne_avg.data[:, :-1]
+    ft_data = avg_ft.data
+
+    check_data(mne_data, ft_data, cur_system)
+    check_info_fields(mne_avg, avg_ft, use_info)
+
+
+@testing.requires_testing_data
+@pytest.mark.filterwarnings('ignore::RuntimeWarning')
+@pytest.mark.parametrize('cur_system, version, use_info',
+                         all_test_params_epochs)
+def test_epoched(cur_system, version, use_info):
+    """Test comparing reading an Epochs object and the FieldTrip version."""
+    has_pandas = _check_pandas_installed(strict=False) is not False
+    test_data_folder_ft = get_data_paths(cur_system)
+    mne_epoched = get_epoched_data(cur_system)
+    if use_info:
+        info = get_raw_info(cur_system)
+        pytestwarning = {'expected_warning': None}
+    else:
+        info = None
+        pytestwarning = no_info_warning
+
+    cur_fname = os.path.join(test_data_folder_ft,
+                             'epoched_%s.mat' % (version,))
+    if has_pandas:
+        pandas = _check_pandas_installed()
```

Comment:
why calling twice _check_pandas_installed in this function?

Reply:
refactored